srlife/damage.py

```diff
@@ -118,6 +126,158 @@ def tube_log_reliability(self, tube, material, receiver):
         # Return the sums as a function of time
         return np.sum(inc_prob, axis=1)
 
+
+class CrackShapeIndependent(WeibullFailureModel):
+    """
+    Parent class for crack shape dependent models
+    """
+
+    def __init__(self, temperatures, material):
+        """
+        Input the material parameters
+        Create a mesh grid for integration
+        Evaluate direction cosines using the mesh grid
+        only for PIA and WNTSA models which are crack shape independent
+        """
+        super().__init__(temperatures, material)
+        self.temperatures = temperatures
+        self.material = material
+        # Material parameters
+        self.svals = self.material.strength(temperatures)
+        self.mvals = self.material.modulus(temperatures)
+        self.kvals = self.svals ** (-self.mvals)
+        self.kpvals = (2 * self.mvals + 1) * self.kvals
+
+        # limits and number of segments for angles
+        self.nalpha = 21
+        self.nbeta = 31
+
+        # Mesh grid of the vectorized angle values
+        self.A, self.B = np.meshgrid(
+            np.linspace(0, np.pi, self.nalpha),
+            np.linspace(0, 2 * np.pi, self.nbeta, endpoint=False),
+            indexing="ij",
+        )
+
+        # Increment of angles to be used in evaluating integral
+        self.dalpha = (self.A[-1, -1] - self.A[0, 0]) / (self.nalpha - 1)
+        self.dbeta = (self.B[-1, -1] - self.B[0, 0]) / (self.nbeta - 1)
+
+        # Direction cosines
+        self.l = np.cos(self.A)
+        self.m = np.sin(self.A) * np.cos(self.B)
+        self.n = np.sin(self.A) * np.sin(self.B)
+
+
+class CrackShapeDependent(WeibullFailureModel):
+    """
+    Parent class for crack shape dependent models
+    """
+
+    def __init__(self, temperatures, material, mandel_stress):
+        """
+        Input the material parameters
+        Create a mesh grid for integration
+        Evaluate direction cosines using the mesh grid
+        for fracture based models which are crack shape dependent
+        """
+        super().__init__(temperatures, material, mandel_stress)
+        self.temperatures = temperatures
+        self.material = material
+        self.mandel_stress = mandel_stress
+        # Material parameters
+        self.svals = self.material.strength(temperatures)
```

Comment:
Anything that depends on `material` `mandel_stress` or `temperature` needs to be calculated in the log reliability function.